apps/librelingo_json_export/librelingo_json_export/export.py

```diff
@@ -20,9 +20,16 @@ def _ensure_output_dir(output_file_path):
     output_file_path.parent.mkdir(parents=True, exist_ok=True)
 
 
-def _save_as_json_file(data, output_path):
-    _ensure_output_dir(output_path)
-    with open(output_path, "w", encoding="utf-8") as f:
+def _prepare_output_path(output_file_path, settings):
+    res = Path(os.devnull)
+    if not _is_dry_run(settings):
+        res = output_file_path
+        _ensure_output_dir(res)
+    return res
+
+
+def _save_as_json_file(data, output_path, settings):
+    with open(_prepare_output_path(output_path, settings), "w", encoding="utf-8") as f:
         json.dump(data, f, ensure_ascii=False, indent=2)
```

Comment:
What strikes me is that this function is specific to JSON files, but it does a logic that is not specific to JSON.

What if instead you just return a file pointer from this function, like so:

```suggestion
def _open_output_file(output_path, settings):
    return open(_prepare_output_path(output_path, settings), "w", encoding="utf-8")
```

Then you could use this for both JSON and YAML files, for example:

```python
    if skill.introduction:
        with _open_output_file(Path(export_path) / "introduction" / f"{slug}.md", settings) as f:
            f.write(skill.introduction)
```

so basically the API is very close to the builtin `open` but it handles the dry run and potentially other features

Reply:
This is a real beauty!